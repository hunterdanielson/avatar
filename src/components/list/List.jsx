import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({ list }) => {
  const mappedList = list.map(item => {
    return <li key={item._id}>
      <h3>{item.name}</h3>
      <p>{item.affiliation}</p>
      <Link to={`/characters/${item._id}`} >Details</Link>
    </li>;
  });

  return <ul>
    {mappedList}
  </ul>;
};

List.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default List;
