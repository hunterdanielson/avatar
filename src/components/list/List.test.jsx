import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List list={[{ _id: 1, name: 'Aang' }, { _id: 2, name: 'Zuko' }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
