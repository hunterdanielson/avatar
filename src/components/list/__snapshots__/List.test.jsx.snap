// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul>
  <li
    key="1"
  >
    <h3>
      Aang
    </h3>
    <p />
    <Link
      to="/characters/1"
    >
      Details
    </Link>
  </li>
  <li
    key="2"
  >
    <h3>
      Zuko
    </h3>
    <p />
    <Link
      to="/characters/2"
    >
      Details
    </Link>
  </li>
</ul>
`;
